### PR TITLE
Modularization: Migrate clientv3 to go.etcd.io/etcd/client/v3

### PR DESCRIPTION
**Pull request by ptabor**
Analogious to go.etcd.io/etcd/client/v2 in previous PR: https://github.com/etcd-io/etcd/pull/12391

We make v3 client code a module go.etcd.io/etcd/client/v3, by dir rename, file move, imports change.

### CODE REVIEW

File: client/v2/go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.15
 require (
 	github.com/json-iterator/go v1.1.10
 	github.com/modern-go/reflect2 v1.0.1
-	go.etcd.io/etcd/api/v3 v3.0.0-20201012212543-0b95e8cef14a
+	go.etcd.io/etcd/api/v3 v3.0.0-00010101000000-000000000000
```
Comment (jingyih) [inline client/v2/go.mod:8]: Not sure why the v2 go.mod and go.sum are changed?

Comment (ptabor) [reply client/v2/go.mod:8]: It was overlooked in previous CLs that not all the versions stay at consistent version. 
When all modules are split, IMHO we should tag them all (so change also here) to v3.5.0-pre

Before: 
```
% git grep go.etcd.io/etcd | grep "v3.0.0" 
client/v2/go.mod:	go.etcd.io/etcd/api/v3 v3.0.0-20201012212543-0b95e8cef14a
client/v2/go.mod:	go.etcd.io/etcd/pkg/v3 v3.0.0-00010101000000-000000000000
go.mod:	go.etcd.io/etcd/api/v3 v3.0.0-20201012212543-0b95e8cef14a
go.mod:	go.etcd.io/etcd/pkg/v3 v3.0.0-00010101000000-000000000000
go.mod:	go.etcd.io/etcd/raft/v3 v3.0.0-00010101000000-000000000000
raft/go.mod:	go.etcd.io/etcd/pkg/v3 v3.0.0-00010101000000-000000000000
tests/go.mod:	go.etcd.io/etcd/api/v3 v3.0.0-20201012212543-0b95e8cef14a
tests/go.mod:	go.etcd.io/etcd/pkg/v3 v3.0.0-00010101000000-000000000000
tests/go.mod:	go.etcd.io/etcd/raft/v3 v3.0.0-00010101000000-000000000000
tests/go.mod:	go.etcd.io/etcd/v3 v3.0.0-00010101000000-000000000000
```
After:
```
% git grep go.etcd.io/etcd | grep "v3.0.0"       
client/v2/go.mod:	go.etcd.io/etcd/api/v3 v3.0.0-00010101000000-000000000000
client/v2/go.mod:	go.etcd.io/etcd/pkg/v3 v3.0.0-00010101000000-000000000000
client/v3/go.mod:	go.etcd.io/etcd/api/v3 v3.0.0-00010101000000-000000000000
client/v3/go.mod:	go.etcd.io/etcd/pkg/v3 v3.0.0-00010101000000-000000000000
go.mod:	go.etcd.io/etcd/api/v3 v3.0.0-00010101000000-000000000000
go.mod:	go.etcd.io/etcd/client/v3 v3.0.0-00010101000000-000000000000
go.mod:	go.etcd.io/etcd/pkg/v3 v3.0.0-00010101000000-000000000000
go.mod:	go.etcd.io/etcd/raft/v3 v3.0.0-00010101000000-000000000000
raft/go.mod:	go.etcd.io/etcd/pkg/v3 v3.0.0-00010101000000-000000000000
tests/go.mod:	go.etcd.io/etcd/api/v3 v3.0.0-00010101000000-000000000000
tests/go.mod:	go.etcd.io/etcd/client/v3 v3.0.0-00010101000000-000000000000
tests/go.mod:	go.etcd.io/etcd/pkg/v3 v3.0.0-00010101000000-000000000000
tests/go.mod:	go.etcd.io/etcd/raft/v3 v3.0.0-00010101000000-000000000000
tests/go.mod:	go.etcd.io/etcd/v3 v3.0.0-00010101000000-000000000000
```

**Review by jingyih — COMMENTED** [review]
LGTM. Thanks!

**Comment by jingyih** [comment]
Is there already an entry in CHANGELOG for the modularization changes? If not, could you help add it?

**Pull request merged**